pyemma/coordinates/clustering/kmeans.py

```diff
@@ -226,23 +226,14 @@ def _estimate(self, iterable, **kw):
                                    format(len(self.clustercenters), self.n_clusters))
 
         # run k-means with all the data
-        it = 0
-        prev_cost = 0
-        with self._progress_context():
-            while it < self.max_iter:
-                self.clustercenters = self._inst.cluster(self._in_memory_chunks, self.clustercenters, self.n_jobs)
-                cost = self._inst.cost_function(self._in_memory_chunks, self.clustercenters, self.n_jobs)
-                rel_change = np.abs(cost - prev_cost) / cost if cost != 0.0 else 0.0
-                prev_cost = cost
-
-                if rel_change <= self.tolerance:
-                    self._converged = True
-                    self._logger.info("Cluster centers converged after %i steps.", it + 1)
-                    break
-                else:
-                    self._progress_update(1, stage=1)
-                it += 1
-            if not self._converged:
+        with self._progress_context(stage=1):
+            if self.show_progress:
+                self._inst.set_callback(lambda: self._progress_update(1, stage=1))
```

Comment:
how about registering the callback as an argument to `cluster_loop`, keeping it localized? I very much like this way of registering the progress update btw